### ISSUE TITLE
v13.0.3 Memory Leak due to Reference Loop on .NET 8

### ISSUE THREAD

**Issue by dlitty** [JamesNK/Newtonsoft.Json#2933]
### Source/destination types

We have database entities for Entity Framework that are being serialized.  This code worked fine with .NET 7, but now that we are upgrading to .NET 8, it is apparently no longer ignoring self-references, even though we still have this code:

```
message_data = JsonConvert.SerializeObject(data, new JsonSerializerSettings() { ReferenceLoopHandling = ReferenceLoopHandling.Ignore}),
```
### Expected behavior

The objects get serialized

### Actual behavior

Memory continues to grow until the OS runs out

### Steps to reproduce

```csharp
See above
```

**Comment by elgonzo** [comment]
> Steps to reproduce
> See above

No, this are not steps to reproduce. That's like saying the recipe for chocolate cookies is: Bake!

**Comment by somunich** [comment]
I have the exact error in .net 8 ReferenceLoopHandling setting is beign ignored, until I get OutOfMemoryException
